### PR TITLE
Various build fixes

### PR DESCRIPTION
**Pull request by bkoropoff**
This fixes build problems I ran into on Fedora 27,

### CODE REVIEW

**Comment by bkoropoff** [comment]
Added some unpacking changes to save bandwidth

**Comment by Kivutar** [comment]
Your idea of using PKG_GIT_URL instead of the libretro PKG_SECTION is good.

Where do the other changes to patches come from?

**Comment by bkoropoff** [comment]
@Kivutar I just wrote the patches by working through the build errors one at a time.

**Comment by ToKe79** [comment]
`PKG_GIT_URL` should be then also added to `config/path`, i.e. set it to empty, so a value from previous package is not re-used with current package if not set in current `package.mk` (in the section after `# set package metadata`). But maybe there is a different reason why all the PKG_* are re-set in config/path.

**Comment by bkoropoff** [comment]
@ToKe79 done

File: packages/libretro/sameboy/package.mk [hunk 1]
```diff
@@ -25,6 +25,7 @@ PKG_ARCH="any"
 PKG_LICENSE="MIT"
 PKG_SITE="https://github.com/libretro/sameboy"
 PKG_URL="https://github.com/libretro/sameboy/archive/$PKG_VERSION.tar.gz"
+PKG_SOURCE_DIR="SameBoy-$PKG_VERSION*"
```
Comment (RobLoach) [inline packages/libretro/sameboy/package.mk]: * ?

Comment (bkoropoff) [reply packages/libretro/sameboy/package.mk]: If you don't specify the name of the directory that gets unpacked from the tarball, it guesses (in this case incorrectly) from the package name.

**Pull request merged**